### PR TITLE
[MAINTENANCE] invoke show-automerges

### PR DESCRIPTION
**Pull request by Kilo59**
Adds a new `show-automerges` invoke task. That can be used to show all the current PRs that are set on auto-merge.
This can be helpful when prioritizing PRs to merge as part of a release. Or just for anyone interested in the currently queued PRs.

The task can be executed by running either of these commands from the project root.
`invoke show-automerges`
`invoke automerge`

image

Thank you @cdkini for the idea and reference script.
```sh
automerge () {
	curl -s https://api.github.com/repos/great-expectations/great_expectations/pulls | jq '.[] | select(.auto_merge.merge_method == "squash") | {title: .title, author: .user.login, date: .created_at, link: .html_url}'
}
```

### Github API docs
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -733,3 +738,30 @@ def link_checker(ctx: Context, skip_external: bool = True):
...
+            "sort": "updated",
+            "direction": "desc",
+            "per_page": 50,
+        },
+    )
+    LOGGER.debug(f"{response.request.method} {response.request.url} - {response}")
+    pr_details = response.json()
+    LOGGER.debug(pf(pr_details, depth=2))
+
+    if automerge_prs := tuple(x for x in pr_details if x["auto_merge"]):
```
Comment (Kilo59) [inline tasks.py:767]: It seemed like just checking for the presence of this key could be enough but @cdkini you probably have more experience here.

File: tasks.py [hunk 2]
```diff
@@ -733,3 +738,30 @@ def link_checker(ctx: Context, skip_external: bool = True):
...
+    url = "https://api.github.com/repos/great-expectations/great_expectations/pulls"
+    response = requests.get(
+        url,
+        params={
+            "state": "open",
+            "sort": "updated",
+            "direction": "desc",
+            "per_page": 50,
+        },
+    )
```
Comment (cdkini) [inline tasks.py:757]: There's a good chance users will forget the set their `GITHUB_TOKEN` env var - we could `raise_for_status` or honestly just keep it as is

Comment (cdkini) [reply tasks.py:757]: No need to overengineer

**Review by cdkini — APPROVED** [review]

File: tasks.py (hunk 1 above)
Comment (cdkini) [reply tasks.py:767]: does this work when you run in locally? If you look at the `jq` command I've written, it is a bit more granular and specifies `squash` as the method - I'm good with this if it works

File: tasks.py (hunk 2 above)
Comment (Kilo59) [reply tasks.py:757]: Oh, yeah let me add the raise for status.

File: tasks.py (hunk 1 above)
Comment (Kilo59) [reply tasks.py:767]: Yes it works locally. I added a screenshot to the PR description.

**Pull request merged**